public/app/plugins/datasource/tempo/SearchTraceQLEditor/SearchField.tsx

```diff
@@ -114,12 +128,16 @@ const SearchField = ({ filter, datasource, updateFilter, deleteFilter, isTagsLoa
           aria-label={`select ${filter.id} scope`}
         />
       )}
-      {filter.type === 'dynamic' && (
+      {!hideTag && (
         <Select
           className={styles.dropdown}
           inputId={`${filter.id}-tag`}
           isLoading={isTagsLoading}
-          options={tags.map((t) => ({ label: t, value: t }))}
+          // Add the current tag to the list if it doesn't exist in the tags prop, otherwise the field will be empty even though the state has a value
+          options={(!filter.tag || tags.find((t) => t === filter.tag) ? tags : [filter.tag, ...tags]).map((t) => ({
```

Comment:
Wouldn't something like `_.uniq([filter.tag, ...tags])` make more sense here? We would not need to do the complicated ternary, and I assume there is not that many value that the perf diff would be noticeable.